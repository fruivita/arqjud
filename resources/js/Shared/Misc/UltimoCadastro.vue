<!--
    Componente destinado a exibição do último registro cadastrado.

    Útil para exibir na tela de cadastro, o último item cadastrado.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import { useTranslationsStore } from '@/Stores/TranslationsStore';

defineProps({
    href: { type: String },
    texto: { type: String },
});

const __ = useTranslationsStore().__;
</script>

<template>
    <div>
        <div class="text-sm text-primaria-700 dark:text-secundaria-300">
            {{ __('Último item cadastrado:') }}
        </div>

        <component
            :class="{
                'underline hover:opacity-80': href,
            }"
            :href="href"
            :is="href ? 'InertiaLink' : 'p'"
            class="font-bold"
            v-html="texto ?? __('Nenhum item cadastrado até o momento')"
        />
    </div>
</template>
