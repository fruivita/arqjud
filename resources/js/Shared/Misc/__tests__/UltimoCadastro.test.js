/**
 * Testes para o componente UltimoCadastro.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import UltimoCadastro from '@/Shared/Misc/UltimoCadastro.vue';
import { mount } from '@vue/test-utils';
import { createPinia, setActivePinia } from 'pinia';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    setActivePinia(createPinia());

    mountFunction = (options = {}) => {
        return mount(UltimoCadastro, { ...options });
    };
});

describe('UltimoCadastro', () => {
    // Caminho feliz
    test('propriedades do componente estÃ£o definidas', () => {
        expect(UltimoCadastro.props).toMatchObject({
            href: { type: String },
            texto: { type: String },
        });
    });

    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { href: 'http://foo.bar', texto: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente sem as suas propriedades respeitando o snapshot', () => {
        expect(mountFunction().html()).toMatchSnapshot();
    });
});
