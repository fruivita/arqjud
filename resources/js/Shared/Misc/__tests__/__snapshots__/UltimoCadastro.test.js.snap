// Vitest Snapshot v1

exports[`UltimoCadastro > renderiza o componente respeitando o snapshot 1`] = `
"<div>
  <div class=\\"text-sm text-primaria-700 dark:text-secundaria-300\\">Último item cadastrado:</div>
  <inertialink class=\\"underline hover:opacity-80 font-bold\\" href=\\"http://foo.bar\\">foo</inertialink>
</div>"
`;

exports[`UltimoCadastro > renderiza o componente sem as suas propriedades respeitando o snapshot 1`] = `
"<div>
  <div class=\\"text-sm text-primaria-700 dark:text-secundaria-300\\">Último item cadastrado:</div>
  <p class=\\"font-bold\\">Nenhum item cadastrado até o momento</p>
</div>"
`;
